hail/python/hail/methods/pca.py

```diff
@@ -315,64 +315,59 @@ def _blanczos_pca(entry_expr, k=10, compute_loadings=False, q_iterations=2, over
     # Generate random matrix G
     G = hl.nd.zeros((n, L)).map(lambda n: hl.rand_norm(0, 1))
 
-    def hailBlanczos(A, G, k, q):
+    def hailBlanczos(A, G, k, q, compute_U=False):
 
-        h_list = []
         G_i = hl.nd.qr(G)[0]
+        g_list = [G_i]
 
         for j in range(0, q):
-            info(f"blanczos_pca: Beginning iteration {j + 1}/{q+1}")
-            temp = A.annotate(H_i=A.ndarray @ G_i)
-            temp = temp.annotate(G_i_intermediate=temp.ndarray.T @ temp.H_i)
-            result = temp.aggregate(hl.struct(Hi_chunks=hl.agg.collect(temp.H_i),
-                                              G_i=hl.agg.ndarray_sum(temp.G_i_intermediate)), _localize=False)._persist()
-            localized_H_i = hl.nd.vstack(result.Hi_chunks)
-            h_list.append(localized_H_i)
-            G_i = hl.nd.qr(result.G_i)[0]
-
-        info(f"blanczos_pca: Beginning iteration {q+ 1}/{q+1}")
-        temp = A.annotate(H_i=A.ndarray @ G_i)
-        result = temp.aggregate(hl.agg.collect(temp.H_i), _localize=False)._persist()
+            info(f"blanczos_pca: Beginning iteration {j+1}/{q}")
+            G_i = A.aggregate(hl.agg.ndarray_sum(A.ndarray.T @ (A.ndarray @ G_i)), _localize=False)
+            G_i = hl.nd.qr(G_i)[0]._persist()
```

Comment:
Why persist individual `G_i`s? Seems like you use them each exactly once.

Reply:
We save them in `g_list` and use them in the next iteration.

Reply:
Duh, you're right. Approved.